src/App.tsx

```diff
@@ -1,26 +1,191 @@
+  const [input, setInput] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [tempDeleteTodo, setTempDeleteTodo] = useState<Todo | null>(null);
+  const [tempClearAll, setTempClearAll] = useState<Todo[] | null>(null);
+
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onError = (errorString: string) => {
+    setErrorMessage(errorString);
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case 'active':
+        return !todo.completed;
+      case 'completed':
+        return todo.completed;
+      case 'all':
+      default:
+        return todo;
+    }
+  });
+
+  const handlClearAll = async () => {
+    try {
+      setTempClearAll(completedTodos);
+      const deletionList = await Promise.allSettled(
+        completedTodos.map(todo => deleteTodo(todo.id)),
+      );
+
+      const hasErorr = deletionList.some(item => item.status === 'rejected');
+      const seccessIds = completedTodos
+        .map((todo, i) =>
+          deletionList[i].status === 'fulfilled' ? todo.id : null,
+        )
+        .filter((id): id is number => id !== null);
+
+      setTodos(prev => prev.filter(todo => !seccessIds.includes(todo.id)));
+      setTempClearAll(null);
+
+      if (hasErorr) {
+        onError('Unable to delete a todo');
+      }
+    } catch {
+      onError('Unable to delete a todo');
+    }
+  };
+
+  const handleDelete = async (todosId: number) => {
+    try {
+      const deletedTodo =
+        filteredTodos.find(todo => todo.id === todosId) || null;
+
+      setTempDeleteTodo(deletedTodo);
+      const deleteAproved = await deleteTodo(todosId);
+
+      if (deleteAproved) {
+        setTodos(filteredTodos.filter(todo => todo.id !== todosId));
+        setTempDeleteTodo(null);
+      }
+    } catch (error) {
+      onError('Unable to delete a todo');
+      setTempDeleteTodo(null);
+    }
+  };
+
+  const handleTodoSubmit = async (e: React.KeyboardEvent<HTMLInputElement>) => {
+    try {
+      if (e.key === 'Enter') {
+        e.preventDefault();
+        const cleanInput = input.trim();
+
+        if (cleanInput !== '') {
+          const tempTodoObject: Todo = {
+            id: 0,
+            title: cleanInput,
+            userId: USER_ID,
+            completed: false,
+          };
+
+          setTempTodo(tempTodoObject);
+          const newTodo = await postTodo(cleanInput);
+
+          if (newTodo) {
+            setTempTodo(null);
+            setTodos([...todos, newTodo]);
+            setInput('');
+            setErrorMessage('');
+          }
+        }
+
+        if (cleanInput === '') {
```

Comment:
```suggestion
        if (!cleanInput) {
```
nit: empty string is a falsy value